src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+    serialSetMode(jetiExBusPort, MODE_RX);
+    return jetiExBusPort != NULL;
+}
+
+
+// Jeti Ex Bus CRC calculations for single byte
+uint16_t updateCRC16( uint16_t crc, uint8_t data )
+{
+    uint16_t ret_val;
+    data ^= (uint8_t)(crc) & (uint8_t)(0xFF);
+    data ^= data << 4;
+    ret_val = ((((uint16_t)data << 8) | ((crc & 0xFF00) >> 8))
+               ^ (uint8_t)(data >> 4)
+               ^ ((uint16_t)data << 3));
+    return ret_val;
+} 
+
+
+// Jeti Ex Bus CRC calculations for a frame
+uint16_t calcCRC16(uint8_t *pt, uint8_t msgLen)
+{
+    uint16_t crc16_data = 0;
+    uint8_t mlen = 0;
+    while(mlen < msgLen) {
+        crc16_data = updateCRC16(crc16_data, pt[mlen]);
+        mlen++;
+    }
+    return(crc16_data);
+}
+
+#ifdef TELEMETRY
+
+// Jeti Ex Telemetry CRC calculations for single byte
+uint8_t updateCRC8( uint8_t crc, uint8_t data )
+{
+    crc ^= data;
+    return (crc ^ (crc << 1) ^ (crc << 2) ^ (0x0e090700 >> ((crc >> 3) & 0x18)));
+} 
+
+// Jeti Ex Telemetry CRC calculations for a frame
+uint8_t calcCRC8(uint8_t *pt, uint8_t msgLen)
+{
+    uint8_t crc8_data = 0;
+    uint8_t mlen = 0;
+    for (mlen = 0; mlen < msgLen; mlen++) {
+        crc8_data = updateCRC8(crc8_data, pt[mlen]);
+    }
+    return(crc8_data);
+}
+
+#endif //TELEMETRY
+
+
+void jetiExBusDecodeFrame(uint8_t *pFrame)
+{
+    uint16_t value;
+    uint8_t frameAddr;
+
+    // Decode header
+    switch (((uint16_t)jetiExBusFrame[EXBUS_HEADER_SYNC] << 8) + ((uint16_t)jetiExBusFrame[EXBUS_HEADER_REQ])){
+
+    case EXBUS_CHANNELDATA_TELEMETRY_REQUEST:                   // not yet specified
+    case EXBUS_CHANNELDATA:
+        for (uint8_t i = 0; i < JETIEXBUS_CHANNEL_COUNT; i++) {
+            frameAddr = JETIEXBUS_HEADER_LEN + i * 2;
+            value = ((uint16_t)pFrame[frameAddr + 1]) << 8;
+            value += (uint16_t)pFrame[frameAddr];
+            // Convert to internal format
+            jetiExBusChannelData[i] = value >> 3;
+        }
+        break;
+    }
+}
+
+void jetiExBusFrameReset(void)
+{
+    jetiExBusFramePosition = 0;
+    jetiExBusFrameLength = JETIEXBUS_MAX_FRAME_SIZE;
+}
+
+
+/*
+  supported:
+  0x3E 0x01 LEN Packet_ID 0x31 SUB_LEN Data_array CRC16      // Channel Data with telemetry request (2nd byte 0x01)
+  0x3E 0x03 LEN Packet_ID 0x31 SUB_LEN Data_array CRC16      // Channel Data forbids answering (2nd byte 0x03)
+  0x3D 0x01 0x08 Packet_ID 0x3A 0x00 CRC16                   // Telemetry Request EX telemetry (5th byte 0x3A)
+  
+  other messages - not supported:
+  0x3D 0x01 0x09 Packet_ID 0x3B 0x01 0xF0 CRC16              // Jetibox request (5th byte 0x3B)
+  ...
+  
+*/
+// Receive ISR callback
+static void jetiExBusDataReceive(uint16_t c)
+{
+    uint32_t now;
+    static uint32_t jetiExBusTimeLast = 0;
+    static uint32_t jetiExBusTimeInterval;
+    static uint16_t crc = 0;
+    
+    // Check if we shall reset frame position due to time
+    now = millis();
+
+    jetiExBusTimeInterval = now - jetiExBusTimeLast;
```

Comment:
jetiExBusTimeInterval should be `int32_t`. Interger overflow rules will handle correctly millis()/micros() overflow